packages/react-component-library/src/components/TabSet/index.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react'
+import uuid from 'uuid'
+
+interface TabSetProps {
+  className?: string
+  tabs: any[]
+}
+
+const TabSet: React.FC<TabSetProps> = ({ className = '', tabs }) => {
+  const [activeTab, setActiveTab] = useState(0)
+
+  return (
+    <article className={`rn-tab-set ${className}`}>
+      <header className="rn-tab-set__head">
+        <nav>
+          <ol className="rn-tab-set__tabs">
+            {tabs.map((tab, index) => {
+              const { title } = tab
+
+              return (
+                <li
+                  key={uuid()}
+                  className={`rn-tab-set__tab-item ${
+                    index === activeTab ? 'is-active' : ''
+                  }`}
+                  data-testid="tab"
+                >
+                  <button
```

Comment:
so.. when you click on a tab it's a button that marks itself as active?

Where's the link... where's the thing that says when I click on a tab, do the thing that tab has been told to do?

Reply:
When you click on a tab it sets some internal state (`activeTab`) to the index of the tab that was clicked. An `is-active` CSS class is then conditionally applied to the appropriate content pane based on this state.

Are you suggesting an expansion of this functionality in order to add custom actions when tabs are clicked? Invoking a custom callback or something.

Reply:
I thought that was the point of the tab, not only that it updates itself but it calls a callback to notify that it's state has changes, so you'd use it to show the correct chunk of content in a page

Reply:
The current implementation isn't just the tabs it also includes hidden panes for the content. I could decouple the tabs from the showing / hiding of the content panes and just make clicking the tabs it invoke a callback instead.

Reply:
This functionality has now been implemented.